File: hub/routes/hs_router/route.go
package hs_router

import (
	"github.com/gin-gonic/gin"
	"github.com/suixinio/headscale-hub/controller/hs_controller"
)

func InitRouteRoutes(r *gin.RouterGroup) gin.IRoutes {
	Group := r.Group("/route")
	controller := hs_controller.NewRouteController()
	Group.GET("/list", controller.List)
	Group.POST("/status", controller.EnableStatus)
	Group.DELETE("/delete", controller.Delete)
	return r
}

File: hub-ui/src/api/headscale/route.js
import request from '@/utils/request'

export function getRouteList() {
  return request({
    url: '/api/headscale/route/list',
    method: 'get'
  })
}

export function enableStatus(data) {
  return request({
    url: '/api/headscale/route/status',
    method: 'post',
    data
  })
}

export function deleteRoute(data) {
  return request({
    url: '/api/headscale/route/delete',
    method: 'delete',
    data
  })
}
